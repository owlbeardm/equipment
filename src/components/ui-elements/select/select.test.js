import React from 'react'
import { shallow } from 'enzyme'

import Select from './select'

const defaultProps = { input: { name: 'the-select' }, label: 'the select', options: ['first-option', 'second-option', 'third-option'] }

describe('rendering testing', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(<Select {...defaultProps} />)
  })
  test('renders without error', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('renders correct number of options (including empty one)', () => {
    const optionNodes = wrapper.find('[data-test="option"]')
    expect(optionNodes.length).toBe(defaultProps.options.length + 1)
  })
})
