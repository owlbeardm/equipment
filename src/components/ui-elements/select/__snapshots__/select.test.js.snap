// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering testing renders without error 1`] = `"<div class=\\"form-group bmd-form-group\\"><label class=\\"control-label bmd-label-static\\" for=\\"the-select\\">the select</label><select name=\\"the-select\\" id=\\"the-select\\" class=\\"form-control\\"><option data-test=\\"option\\"></option><option data-test=\\"option\\">first-option</option><option data-test=\\"option\\">second-option</option><option data-test=\\"option\\">third-option</option></select></div>"`;
